test/unit/BCoWPool.t.sol

```diff
@@ -52,10 +58,77 @@ contract BCoWPool_Unit_Finalize is BaseCoWPoolTest {
     _tokensLength = bound(_tokensLength, MIN_BOUND_TOKENS, MAX_BOUND_TOKENS);
     _setRandomTokens(_tokensLength);
     address[] memory tokens = _getDeterministicTokenArray(_tokensLength);
-    for (uint256 i = 0; i < bPool.getNumTokens(); i++) {
+    for (uint256 i = 0; i < bCoWPool.getNumTokens(); i++) {
       vm.mockCall(tokens[i], abi.encodePacked(IERC20.approve.selector), abi.encode(true));
       vm.expectCall(tokens[i], abi.encodeCall(IERC20.approve, (vaultRelayer, type(uint256).max)), 1);
     }
-    bPool.finalize();
+    bCoWPool.finalize();
+  }
+}
+
+/// @notice this tests both commit and commitment
+contract BCoWPool_Unit_Commit is BaseCoWPoolTest {
+  function test_revertOnNonSolutionSettler(address sender, bytes32 orderHash) public {
+    vm.assume(sender != cowSolutionSettler);
+    vm.prank(sender);
+    vm.expectRevert(IBCoWPool.CommitOutsideOfSettlement.selector);
+    bCoWPool.commit(orderHash);
+  }
+
+  function test_setCommitment(bytes32 orderHash) public {
+    vm.prank(cowSolutionSettler);
+    bCoWPool.commit(orderHash);
+    assertEq(bCoWPool.commitment(), orderHash);
+  }
+}
+
+contract BCoWPool_Unit_DisableTranding is BaseCoWPoolTest {
+  function test_revertOnNonController(address sender) public {
+    // contract is deployed by this contract without any pranks
+    vm.assume(sender != address(this));
+    vm.prank(sender);
+    vm.expectRevert(IBPool.BPool_CallerIsNotController.selector);
+    bCoWPool.disableTrading();
+  }
+
+  function test_wipesAppdataHash(bytes32 appDataHash) public {
+    vm.assume(appDataHash != bytes32(0));
+    bCoWPool.set_appDataHash(appDataHash);
+    bCoWPool.disableTrading();
+    assertEq(bCoWPool.appDataHash(), bytes32(0));
+  }
+
+  function test_emitTradingDisabledEvent() public {
+    vm.expectEmit();
+    emit IBCoWPool.TradingDisabled();
+    bCoWPool.disableTrading();
+  }
+
+  function test_succeedOnAlreadyZeroAppdata() public {
+    bCoWPool.set_appDataHash(bytes32(0));
+    bCoWPool.disableTrading();
+  }
```

Comment:
we should also succeed in non-zero appData, idk if it's a feature yet, but if we decide to block it we should have a test that replicates behaviour